app/state.go

```diff
@@ -34,6 +35,49 @@ const (
 		"by currently installed artifact (%v != %v)."
 )
 
+type timerPool sync.Pool
```

Comment:
Nothing against this change, but the performance impact was probably more important before the nanosecond fix! 😀

Reply:
Yeah, it was just a random thing I came across as I traced the code for the unit test. I was contemplating leaving it out of the PR, but decided to leave it in anyway.